app/controllers/users_controller.rb

```diff
@@ -0,0 +1,52 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+
+    if @user.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def login_form
+    @user = User.new
+  end
+
+  def login
+    username = params[:user][:username]
+    user = User.find_by(username: username)
+
+    if user
+      session[:user_id] = user.id
+      flash[:status] = :success
+      flash[:message] = "Successfully logged in as returning user #{username}"
+    else
+      user = User.create(username: username)
+      session[:user_id] = user.id
+      flash[:status] = :success
+      flash[:message] = "Successfully created new user #{username} with ID #{user.id}"
+    end
+
+    redirect_to root_path
+  end
+
+  def current
+    @current_user = User.find_by(id: session[:user_id])
```

Comment:
Finding a logged in user is good functionality to put in a controller filter because it may be used multiple places in the future.